### PR TITLE
Update starknet counter example for Sepolia (devnet and testnet)

### PR DESCRIPTION
**Pull request by devnet0x**
Fix Starknet Counter example for devnet and sepolia networks. Close issue #399 

## Summary by CodeRabbit

- **New Features**
	- Updated configuration for the StarkNet provider to use a local server.
	- Changed the test network identifier to a new value.

- **Bug Fixes**
	- Improved UI responsiveness by altering the control flow in the counter update method.

- **Chores**
	- Updated dependency sources from Git repositories to local paths for better project management.
	- Adjusted default versioning format in transaction initialization.
	- Modified development environment configurations for local network access.

### CODE REVIEW

File: examples/starknet_counter/lib/ui/counter.dart [hunk 1]
```diff
@@ -30,8 +30,8 @@ class _CounterPageState extends State<CounterPage> {
 
   _decreaseCount() async {
     await decreaseCounter();
-    setState(() async {
-      await _getCounter();
+    setState(() {
+      _getCounter();
     });
```
[the remark this answers is not in the document]
Comment (devnet0x) [reply examples/starknet_counter/lib/ui/counter.dart]: Ok, PR updated

File: examples/starknet_counter/lib/services/counter_service.dart [hunk 2]
```diff
@@ -5,19 +5,18 @@ import 'package:starknet_provider/starknet_provider.dart';
...
 final contractAddress =
-    '0x716ea2ba1dc8f1e7f9faad442a109adebe4a80a2ec0c937c7e84aca58136859';
-// final contractAddress ='0x076b4f19561a3c48f13aa3cb912ad92c0e702270466668ec9d77513ba6c5b0e2';
+    '0x0662d4b526b2a0e87a449263b131e1831aa166f2d3d97412e8dc67fb85a3de1c';
 final secretAccountAddress =
-    "0x00ce7b8175e1aed7e087f44e63051c053cda012d5f63fdd1e95e82489925ff41";
+    "0x64b48806902a367c8598f4f95c305e8c1a1acba5f082d294a43793113115691";
 final secretAccountPrivateKey =
-    "0x06a1b5d41b7e5fee4310fda61d7c1b11e039f4681424cc89e1e8bfffe1ed9926";
+    "0x71d7bb07b9a64f6f78ac4c816aff4da9";
```
[the remark this answers is not in the document]
Comment (devnet0x) [reply examples/starknet_counter/lib/services/counter_service.dart:14]: These are well known keys from devnet

**Comment by ptisserand** [comment]
Thanks a lot for your contribution!
Could you please fix the failed signer test: I think it's related to your fix on invoke transaction version.

**Comment by devnet0x** [comment]
Fixed. It was related to a hardcoded signature in the test file (signed with SN_GOERLI instead of SN_SEPOLIA). Algo fixed the melos test devnet port because test was looking for devnet in 5050 and setup was loading devnet in 3030.

image

**Comment by ptisserand** [comment]
Thanks for your contribution

**Pull request merged**